### PR TITLE
add GSoC 2024 proposal

### PR DESCRIPTION
**Pull request by sudiptob2**
Added one proposal for GSOC 2024. 
This idea was originally proposed in GSOC 2023. I copied most of the text from the previous year's idea. It might be required to update some details. Need, experience mainers help with that.
Also, it needs to be decided, who will be able to mentor in the project. 
I plan to submit a proposal as a mentee for this idea in GSOC 2024.

### CODE REVIEW

File: mentorship/gsoc/2024/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Google Summer of Code 2023
```
Comment (mowies) [inline mentorship/gsoc/2024/README.md:1]:
```suggestion
# Google Summer of Code 2024
```

**Comment by mowies** [comment]
would be great to get this approved and merged @thschue @salaboy @AnaMMedina21 :)

**Review by mowies — APPROVED** [review]

**Review by thschue — APPROVED** [review]
Thank you for adding this proposal!

LGTM

**Comment by mowies** [comment]
i guess we can close this since it's outdated

**Pull request closed without merging**